networking/p2p/src/main/java/tech/pegasys/teku/networking/p2p/libp2p/LibP2PPeer.java

```diff
@@ -79,6 +84,34 @@ public LibP2PPeer(
                     "Peer {} connection close future completed exceptionally", peerId, error));
   }
 
+  @Override
+  public void checkPeerIdentity() {
+    if (isConnected()) {
+      getAgentVersionFromIdentity()
+          .thenAccept(
+              maybeAgent -> {
+                LOG.debug("Connected peer has agent string: {}", maybeAgent.orElse("Unknown"));
+                maybeAgentString = maybeAgent;
+                if (maybeAgent.isPresent()) {
+                  peerClientType = getPeerTypeFromAgentString(maybeAgent.get());
+                }
+              })
+          .finish(error -> LOG.error("Failed to retrieve client identity", error));
+    }
```

Comment:
I suspect we can still get a race condition here where we disconnect just as we're trying to retrieve the client identity.

I probably wouldn't bother checking if we're connected and reduce the error log to debug - we don't really care if we can't retrieve the client identity.